src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -42,19 +54,21 @@ public CompanyProfileService(Logger logger, CompanyProfileRepository companyProf
      */
     public void updateInsolvencyLink(final CompanyProfile companyProfileRequest)
             throws NoSuchElementException {
-        Optional<CompanyProfileDocument> companyProfileOptional = companyProfileRepository
-                .findCompanyProfileDaoByCompanyProfile_Data_CompanyNumber(
-                        companyProfileRequest.getData().getCompanyNumber());
+        String companyNumber = companyProfileRequest.getData().getCompanyNumber();
+        Query updateCriteria = new Query(Criteria.where("data.company_number").is(companyNumber));
+        Update updateQuery = new Update();
+        updateQuery.set("data.links.insolvency",
+                companyProfileRequest.getData().getLinks().getInsolvency());
```

Comment:
I think we also need to update etag, updatedBy along with insolvency links element? I would probably prefer to fetching whole object once and persist object with all updated attributes maybe

Reply:
I wasn't aware of updating etag/updatedBy, would it be better to update the update query to include these two fields, to keep it as a single DB call